src/components/ar-paint-controls.js

```diff
@@ -0,0 +1,155 @@
+AFRAME.registerSystem('ar-paint-controls', {
+  numberStrokes: 0
+});
+
+/* globals AFRAME THREE */
+AFRAME.registerComponent('ar-paint-controls', {
+  dependencies: ['brush', 'raycaster'],
+
+  init: function () {
+    var el = this.el;
+    this.controller = null;
+
+    this.size = el.sceneEl.renderer.getSize();
+    this.pointer = new THREE.Vector2();
+    // normalized device coordinates position
+    this.pointerNdc = new THREE.Vector2();
+
+    this.numberStrokes = 0;
+
+    this.raycaster = el.components.raycaster.raycaster;
+    // this.el.components.raycaster.showLine = true;
+    this.ray = this.raycaster.ray;
+    this.intersection = null;
+
+    el.object3D.visible = false;
+
+    var soundEl = document.createElement('a-sound');
+    var iOSSuffix = '';
+    if (AFRAME.utils.device.isIOS()) {
+      iOSSuffix = '_iOS';
+    }
+    soundEl.setAttribute('src', '#ui_paint' + iOSSuffix);
+    soundEl.setAttribute('id', 'uiPaint');
+    this.el.appendChild(soundEl);
+
+    this.onStrokeStarted = this.onStrokeStarted.bind(this);
+    document.querySelector('[ar-ui]').addEventListener('activate', this.activate.bind(this));
```

Comment:
Are these events  always active? Or do we need to remove them sometimes?

Reply:
'activate' / 'deactivate' events from 'ar-ui' are used when enter / exit on a-painter mode. Not sure if we need to remove them or when.

Reply:
Typically we should remove events that are not used anymore when some functionality or tool is deactivated, but not big deal, we could go back to this later.